### PR TITLE
Finalize gmp8 part 1

### PR DESCRIPTION
**Pull request by bjoernricks**
* Avoid duplicate code in tests
* Add get_vulnerabilities, get_vulnerability
* Add create_ticket, modify_ticket, delete_ticket, get_ticket, get_tickets and clone_ticket
* Introduce a CredentialType enum for gmpv8
* Expect CredentialType as argument for create_credential and modify_credential in gmpv8

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [x] Documentation

### CODE REVIEW

File: gvm/protocols/gmpv8.py [hunk 1]
```diff
@@ -541,3 +546,219 @@ def modify_tag(
                 _xmlresources.add_element("type", resource_type)
 
         return self._send_xml_command(cmd)
+
+    def clone_ticket(self, ticket_id):
+        """Clone an existing ticket
+
+        Arguments:
+            result_id (uuid): Result the ticket applies to
+            assigned_to_user_id (uuid): Use the ticket is assigned to
```
Comment (wiegandm) [inline gvm/protocols/gmpv8.py]:
```suggestion
            assigned_to_user_id (uuid): User the ticket is assigned to
```

File: gvm/protocols/gmpv8.py [hunk 2]
```diff
@@ -541,3 +546,219 @@ def modify_tag(
...
+            open_note (str): A note about opening the ticket
+            comment (str, optional): Comment for the target
```
Comment (wiegandm) [inline gvm/protocols/gmpv8.py]:
```suggestion
            comment (str, optional): Comment for the ticket
```

File: gvm/protocols/gmpv8.py [hunk 3]
```diff
@@ -541,3 +546,219 @@ def modify_tag(
...
+
+    def clone_ticket(self, ticket_id):
+        """Clone an existing ticket
+
+        Arguments:
+            result_id (uuid): Result the ticket applies to
+            assigned_to_user_id (uuid): User the ticket is assigned to
+            open_note (str): A note about opening the ticket
+            comment (str, optional): Comment for the ticket
+
```
Comment (wiegandm) [inline gvm/protocols/gmpv8.py]: On a closer look, the last two arguments `open_note` and `comment` either do not exist or are not implemented. Did not notice that in my earlier suggestion ...

File: gvm/protocols/gmpv8.py [hunk 4]
```diff
@@ -541,3 +546,219 @@ def modify_tag(
...
+        if comment:
+            cmd.add_element("comment", comment)
+
+        return self._send_xml_command(cmd)
+
+    def delete_ticket(self, ticket_id, *, ultimate=False):
+        """Deletes an existing ticket
+
+        Arguments:
+            ticket_id (str) UUID of the ticket to be deleted.
```
Comment (wiegandm) [inline gvm/protocols/gmpv8.py]: Is there a reason for documenting the type of `ticket_id` as `str`? `create_ticket` above documents `id`s as `uuid`, this should be consistent IMHO.

There are other occurences below.

File: gvm/protocols/gmpv8.py [hunk 5]
```diff
@@ -541,3 +546,219 @@ def modify_tag(
...
+        comment=None
+    ):
+        """Modify a single ticket
+
+        Arguments:
+            ticket_id (str): UUID of an existing ticket
+            status (TicketStatus, optional): New status for the ticket
+            note (str, optional): Note for the status change. Required if status
+                is set.
+            assigned_to_user_id (uuid): User the ticket is assigned to
```
Comment (wiegandm) [inline gvm/protocols/gmpv8.py]: Looks to be optional.

File: gvm/protocols/gmpv8.py (hunk 3 above)
Comment (bjoernricks) [reply gvm/protocols/gmpv8.py]: You are right. Was a copy&paste mistake.

File: gvm/protocols/gmpv8.py (hunk 4 above)
Comment (bjoernricks) [reply gvm/protocols/gmpv8.py]: Just double checked with the gmpv7 part and we use str as type for uuids currently. Therefore I've adjusted the docstrings accordingly.

File: gvm/protocols/gmpv8.py (hunk 5 above)
Comment (bjoernricks) [reply gvm/protocols/gmpv8.py]: Fixed.

**Pull request merged**